doctor/docs/base.py

```diff
@@ -121,10 +121,101 @@ def get_example_lines(headers: Dict[str, str], method: str, url: str,
     return lines
 
 
+def get_object_reference(obj: Object) -> str:
```

Comment:
I just made this it's own function as we duplicated this logic in 3 or 4 places.